src/databricks/labs/ucx/aws/access.py

```diff
@@ -359,8 +361,12 @@ def create_uber_principal(self, prompts: Prompts):
             self._update_sql_dac_with_instance_profile(iam_instance_profile, prompts)
             logger.info(f"Cluster policy \"{cluster_policy.name}\" updated successfully")
         except PermissionError:
+            logger.error(f"Failed to assign instance profile to cluster policy {iam_role_name}")
             self._aws_resources.delete_instance_profile(iam_role_name, iam_role_name)
```

Comment:
Does this also delete the created roles?

Reply:
It does